src/Sylius/Bundle/ApiBundle/DataTransformer/ShipmentIdAwareInputCommandDataTransformer.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\DataTransformer;
+
+use Sylius\Bundle\ApiBundle\Command\ShipmentIdAwareInterface;
+use Sylius\Component\Core\Model\ShipmentInterface;
+
+/** @experimental */
+final class ShipmentIdAwareInputCommandDataTransformer implements CommandDataTransformerInterface
+{
+    public function transform($object, string $to, array $context = [])
+    {
+        /** @var ShipmentInterface $shipment */
+        $shipment = $context['object_to_populate'];
+
+        $object->setShipmentId((string) $shipment->getId());
```

Comment:
`$id` of shipment can be an integer, we shouldn't cast that value to string
```suggestion
        $object->setShipmentId($shipment->getId());
```